### PR TITLE
Restore normalize allowing passthrough of zero vec

### PR DESCRIPTION
**Pull request by sgraham**
Possible fix for #160 (not sure if there's a more efficient way of doing this).

### CODE REVIEW

**Comment by sgraham** [comment]
Oops, some builds don't like that. I should find a place to put a test too.

File: dev/src/builtins.cpp [hunk 1]
```diff
@@ -710,7 +710,13 @@ nfr("normalize", "vec",  "F}" , "F}",
         VECTORVARS;
         VECTOROPNR(sql += f.fval() * f.fval());
         double m = sqrt(sql);
-        VECTOROPNR(elems[i] = f.fval() / m);
+        if (m == 0.0) {
+            for (iint i = 0; i < len; i++) {
```
Comment (aardappel) [inline dev/src/builtins.cpp:714]: `VECTOROPNR(elems[i] = 0.0)` ?

Comment (sgraham) [reply dev/src/builtins.cpp:714]: That's what I did at first, but the gcc/clang complained that ```/home/runner/work/lobster/lobster/dev/src/builtins.cpp:584:14: error: variable ‘f’ set but not used [-Werror=unused-but-set-variable]
  584 |         auto f = elems[i]; \```

Maybe `VECTOROPNR((void)f; elems[i] = 0.0)` if you prefer that?

Comment (aardappel) [reply dev/src/builtins.cpp:714]: yeah, that's not great, but still better to be consistent.

Comment (sgraham) [reply dev/src/builtins.cpp]: Done

**Pull request merged**